### ISSUE TITLE
header include generated for unused imported variables

### ISSUE THREAD

**Issue by alaviss** [nim-works/nimskull#1125]
### Example

`x.nim`:

```nim
var x* {.importc, header: "does-not-exist".}: int
```

`test.nim`:

```nim
import x
```

### Actual Output

```
$ nim c test.nim

@mx.nim.c:8:10: fatal error: does-not-exist: No such file or directory
```

### Expected Output

Success

### Additional Information

* Found via Windows CI docgen for `posix_utils.nim`
* I'm not sure if this can even be classified as a bug

**Comment by saem** [comment]
It's exported so we have to generate it, then the header doesn't exist, which we shouldn't silently fail. Best I can think of, is if it's an optional item, then we should have a facility to check that via conditional compilation. At which point the module needs fixing to guard with that.

**Comment by alaviss** [comment]
The big issue is how it interacts with `runnableExamples`. I'm fine with failing but it does mean that docgen can't run for those on other OS.

**Comment by saem** [comment]
> The big issue is how it interacts with `runnableExamples`. I'm fine with failing but it does mean that docgen can't run for those on other OS.

Good point, I think `runnableExamples` needs an optional condition that can be provided to it, to conditionally compile code while keeping the example itself clean. How's that sound?

**Comment by alaviss** [comment]
Sounds good to me

**Comment by zerbina** [comment]
Fixed by #1191, but wasn't auto-closed.
